src/DGtal/shapes/MeshVoxelizer.h

```diff
@@ -109,15 +109,111 @@ namespace DGtal
     {
       PointR3 myFirst, mySecond;
       Edge() {}
-      Edge(PointR3 mf, PointR3 ms) : myFirst(mf), mySecond(ms) {}
+      Edge(const PointR3& mf, const PointR3& ms) : myFirst(mf), mySecond(ms) {}
+      Edge(const Edge&) = default;
+    };
+
+    /**
+     * Internal Intersection Target structure
+     *
+     * @tparam Dimension dimension of the intersection target
+     * @tparam Separation strategy of the voxelization (6 or 26)
+     * @tparam Dummy dummy parameter just to be able to specialize inside template class (FIXME with another location)
+     */
+    template<size_t Sep, size_t Dimension = 1, typename Dummy = void>
+    struct IntersectionTarget { };
+
+    /**
+     * Specialization for 6-separated with 1D intersection target
+     */
+    template<typename Dummy>
+    struct IntersectionTarget<6, 1, Dummy>
+    {
+      const std::array<Edge, 3> myTarget {{
+          { { 0.5, 0.0, 0.0 }, { -0.5,  0.0,  0.0} } , // AB
+          { { 0.0, 0.0, 0.5 }, {  0.0,  0.0, -0.5} } , // CD
+          { { 0.0, 0.5, 0.0 }, {  0.0, -0.5,  0.0} }   // EF
```

Comment:
sorry I'm not really good about floating-point arithmetic, I didn't understand this point.
If I put `1/2` instead of `0.5` here, it will be a integer division, so 0, then it will be cast to a double, so `0.0`, and I want `0.5`, no?